lisa/sut_orchestrator/azure/common.py

```diff
@@ -1036,6 +1048,40 @@ def delete_file_share(
     share_service_client.delete_share(file_share_name)
 
 
+def save_console_log(
+    resource_group_name: str,
+    vm_name: str,
+    platform: "AzurePlatform",
+    log: Logger,
+    saved_path: Optional[Path],
+    screenshot_file_name: str = "serial_console",
+) -> bytes:
+    diagnostic_data = _get_diagnostic_data(platform, resource_group_name, vm_name)
+    if saved_path:
+        screenshot_raw_name = saved_path.joinpath(f"{screenshot_file_name}.bmp")
+        screenshot_name = saved_path.joinpath(f"{screenshot_file_name}.png")
+        screenshot_response = requests.get(diagnostic_data.console_screenshot_blob_uri)
```

Comment:
the response from `_get_diagnostic_data()` is manipulated twice before it's consumed and not used anywhere else, so why not move the call to requests and the content lookup below to `_get_diagnostic_data()` where it will be easy to add additional validation?

Reply:
sorry, I didn't catch this, could you please explain it?

Reply:
You created `_get_diagnostic_data()` to move logic out of this function, but then you have
`screenshot_response = requests.get(diagnostic_data.console_screenshot_blob_uri)` and `screenshot_response.content`. 
`_get_diagnostic_data()` hardly does anything, so why not rename it, `get_screenshot()` and have it call requests and only return the content? Then you can add additional error checking in case the requests call fails?

Reply:
Oh, sorry, I missed you were using it below. I would drop _get_diagnostic_data since it's only a couple lines of code and just merge it in here. You also need to add error checking for this requests call.